src/DIRAC/TransformationSystem/DB/TransformationDB.py

```diff
@@ -1617,6 +1620,23 @@ def addFile(self, fileDicts, force=False, connection=False):
             gLogger.info("Files to add to transformations:", filesToAdd)
             if filesToAdd:
                 for transID, lfns in transFiles.items():
+                    # Check if file is already in the Transformation and if is in Deleted Status
+                    res = self.getTransformationFiles(
+                        condDict={"TransformationID": transID, "LFN": lfns}, connection=connection
+                    )
+                    if not res["OK"]:
+                        return res
+                    fileIDs = []
+                    for fileDict in res["Value"]:
+                        fileIDs.append(fileDict["FileID"])
+                        if fileDict["Status"] == "Deleted":
+                            res = self.__setTransformationFileStatus(list(fileIDs), "Unused", connection=connection)
```

Comment:
fileIDs is appended to inside the loop, probably one should only use the current fileID?

Reply:
Or maybe split the fileIDs in Deleted and New and call once the status?

Reply:
> fileIDs is appended to inside the loop, probably one should only use the current fileID?

Yes I agree. 
What about removing :

`fileIDs.append(fileDict["FileID"])`

and modifying in this way :

`list([fileDict["FileID"]]), TransformationFilesStatus.UNUSED, connection=connection`

?